scripts/sexbound/plugins/climax/inflation.lua

```diff
@@ -0,0 +1,137 @@
+Inflation = {}
+Inflation_mt = {
+    __index = Inflation
+}
+
+function Inflation:new(parent, config)
+    local _self = setmetatable({
+        _logPrefix = "INFL",
+        _parent = parent,
+        _config = config,
+        _actor = {},
+        _log = {},
+        _defaultLiquidId = config.projectileLiquid["default"]["male"] or "semen",
+        _loads = {},
+        _loadCount = 0,
+        _totalInflation = 0
+    }, Inflation_mt)
+
+    _self._log = Sexbound.Log:new(_self._logPrefix, _self._parent:getRoot():getConfig())
+    _self._actor = _self._parent._parent
+    _self:getLog():info("Inited Inflation Tracker for ".._self:getActorLogPrefix())
+
+    return _self
+end
+
+function Inflation:addLoad(inflationLoad)
+    inflationLoad = {
+        liquid = inflationLoad.liquid or self._defaultLiquidId,
+        quantity = inflationLoad.quantity or 0
+    }
+    if inflationLoad.quantity <= 0 then return end
+
+    self:getLog():debug(self:getActorLogPrefix().." received inflation load. "..
+                        "Liquid: ["..inflationLoad.liquid.."] "..
+                        "Quantity: ["..inflationLoad.quantity.."]")
+
+    -- Add to the last load if it has the same liquid id
+    if self._loadCount > 0 and self._loads[self._loadCount].liquid == inflationLoad.liquid then
```

Comment:
There is a split point here.

Option A: In case we continue to just track the different liquid types there is no need to make a cocktail stack and only ever check the last layer for the same liquid - just iterate over all different stored liquids and add to the pile that fits, if any does.

Option B: Since you already do keep track of strongly seperated loads, might as well future-proof it by also tracking which entityId the liquid belongs to. In that case the "liquid" equality check is redundant, as the same entity should only ever produce the same liquid, and that information can later be used when the system gets expanded to track "stored" liquids even outside of the sexnode, on the actual entity. Could be used for more realistic means of post-sex impregnation and other effects, like dripping.

Reply:
Appendix:
Yes, the first approach would naturally require to change up the "remove" logic - probably just a simple "determine percentage that is to be removed, then remove that from all stacks. Anything <0.01 gets removed entirely."